File: 2630-alternating-digit-sum/2630-alternating-digit-sum.py
class Solution:
    def alternateDigitSum(self, n: int) -> int:
        sign = 0
        ans = 0
        digits = 0
        while n:
            digit = n % 10
            n //= 10
            digits += 1
            if sign:
                ans -= digit
            else:
                ans += digit
            
            sign = 1 - sign

        if digits % 2 == 0:
            return -ans
        return ans

File: 2630-alternating-digit-sum/2630-alternating-digit-sum.cpp
class Solution {
public:
    int alternateDigitSum(int n) {
        int digits = 0;
        int sum = 0;
        int sign = 0;

        while (n) {
            int digit = n % 10;
            ++digits;
            if (sign) {
                sum -= digit;
            } else {
                sum += digit;
            }
            sign = 1 - sign;
            n /= 10;
        }

        if (digits % 2 == 0) {
            return -1 * sum;
        }

        return sum;
    }
};